honeybee_vtk/tovtk.py

```diff
@@ -0,0 +1,311 @@
+"""Functions to create vtk objects based on the data extracted from hbjson."""
+
+import vtk
+
+
+def group_by_face_type(points_lst, hb_types):
+    """Group points based on Honeybee type.
+
+    Args:
+        points_lst: A list of lists of points for the geometry in the HBJSON. Here,
+            each point is a list of X, Y, and Z cordinates of the point.
+        hb_types: A list of honeybee face_types and/or types for the geometry objects
+            in the HBJSON.
+
+    Returns:
+        A dictionary with Honeybee type as keys and list of list of points for geometry
+        that belongs to that Honeybee type.
+    """
+    grouped_points = {face_type: [] for face_type in hb_types}
+
+    for i in range(len(hb_types)):
+        grouped_points[hb_types[i]].append(points_lst[i])
+
+    return grouped_points
+
+
+def create_polygon(points):
+    """Create a vtk Polygon from a list of points.
+
+    The point will have to be in an order. This function cannot be used to create a
+    polygon with holes.
+
+    Args:
+        points: A list of points. Here, each point is a list of X, Y, and Z cordinate.
+
+    Returns:
+        A tuple with two elements.
+
+        - vtk_points: A list of vtk point objects
+
+        - vtk_polygon: A vtk polygon object created from the points provided.
+    """
+    vtk_points = vtk.vtkPoints()
+    vtk_polygon = vtk.vtkPolygon()
+    vtk_polygon.GetPointIds().SetNumberOfIds(len(points))
+
+    for point in points:
+        vtk_points.InsertNextPoint(tuple(point))
+
+    for i in range(len(points)):
+        vtk_polygon.GetPointIds().SetId(i, i)
+
+    return vtk_points, vtk_polygon
+
+
+def create_polygons(points_lst):
+    """Create a vtk object with multiple vtk polygons.
+
+    Args:
+        points_lst: A list of lists of points. Here, each point is a list of X, Y, 
+            and Z cordinate.
+
+    Returns:
+        A vtk object that with multiple polygons.
+    """
+    vtk_polydata_lst = []
+
+    for i in range(len(points_lst)):
+
+        vtk_points, vtk_polygon = create_polygon(points_lst[i])
+        vtk_polygons = vtk.vtkCellArray()
+        vtk_polygons.InsertNextCell(vtk_polygon)
+
+        vtk_polydata = vtk.vtkPolyData()
+        vtk_polydata.SetPoints(vtk_points)
+        vtk_polydata.SetPolys(vtk_polygons)
+        vtk_polydata.Modified()
+
+        vtk_polydata_lst.append(vtk_polydata)
+
+    vtk_polydata_extended = vtk.vtkAppendPolyData()
+    
+    for i in range(len(vtk_polydata_lst)):
```

Comment:
Almost every time that you write `range(len(some_list))` there is probably an easier way to do it in Python including here.